### DISCUSSION TITLE
Installed on new machine, pulled only db, now attachment writes fail to wrong dir

### DISCUSSION THREAD

**Discussion by dglidden** [mastodon/mastodon#33413] [Q&A] [answered]
As the title says, I started over on a new machine (more power, more storage) than the one that had been running Mastodon.  The only things I pulled over is the database dump as of the last request after shutting down the old instance and the .env.production with the various keys, but started over with everything else as per the "Installing Mastodon From Scratch" document.  The old install was under /opt/mastodon, the new one is under /home/mastodon/live but the attachments are still trying to be saved in /opt/mastodon.   Since the database and .env.production are the only things that moved over I have to assume the storage option is somewhere in there, but I can't find it in the database, and obviously if it's encoded in one of the keys, it's not something I can get to.  Setting PAPERCLIP options in .env.production doesn't seem to help.  In the meantime I've tried to tell systemd to let /opt/mastodon be in the ReadWriteDirs for the mastodon processes, but that doesn't seem to be working either.   I've dug through the database, and as much of the code as I can work out in a couple of hours, and found nothing else.  Everything else seems to be working "just fine" in that I'm getting feeds and able to post, but of course, images are all broken. Help?

**Comment by noellabo** [comment]
Start the rails console as the mastodon user.

Type `ENV` to check the environment variables.

```
mastodon@new:~/live$ RAILS_ENV=production bin/rails c

irb(main):001> ENV
```

Is there a variable pointing to '/opt/mastodon'?

Can't find the problem? OK, let's try saving the attachment appropriately.

Continuing with the rails console. I will omit the prompt.

```
Rails.logger = Logger.new(STDOUT)
m = MediaAttachment.new
m.account = Account.find_local('admin')
m.remote_url='https://docs.joinmastodon.org/assets/network-models.jpg'
m.download_file!
m.save!
```

If you can get and save the file without any errors, there is no problem with the installation such as `.env.producion`.
Suspect the systemd unit files (mastodon-web.service, mastodon-sidekiq.service).

**Reply by dglidden** [reply, marked as answer]
It turned out indeed to be the systemd unit files, but probably not what one might immediately suspect.  Somehow they got symlinked to the old content directory where I had copied everything off of the old server.  When I removed the old content dir, systemd could no longer find the startup files which led me to this discovery.  The systemd files in the wrong place let everything work except somehow Rails was finding the OLD content directory through startup, despite the startup files pointing at the new install as the working directory.  It was then running it from the new install, but file access was being pointed at the old directory through some kind of systemd magic.  When I put the proper symlinks in, everything worked, minus a few standard permission issues.

I appreciate the hints, and I'm glad I was able to finally get it working.
